### PR TITLE
Transfer ownership of Router in RouterDeployer

### PR DESCRIPTION
**Pull request by nambrot**
This PR points against #537 but should be repointed once that is merged.

This PR transfers ownership of the router in RouterDeployer according to the config. It allows subclasses to define what transferring ownership really means (for the controller the semantics are slightly different)

### CODE REVIEW

File: typescript/deploy/src/router/deploy.ts [hunk 1]
```diff
@@ -57,6 +55,24 @@ export abstract class AbacusRouterDeployer<
         }
       }),
     );
+  }
+
+  async transferOwnership(contractsMap: ChainMap<Chain, Contracts>) {
+    this.logger(`Transfer Ownership of routers to owner ...`);
```
Comment (asaj) [inline typescript/deploy/src/router/deploy.ts:61]: lowercase `ownership`

File: typescript/deploy/src/router/deploy.ts [hunk 2]
```diff
@@ -39,9 +39,7 @@ export abstract class AbacusRouterDeployer<
     return router instanceof ProxiedContract ? router.contract : router;
   }
 
-  async deploy() {
-    const contractsMap = await super.deploy();
-
+  async mututallyEnrollRouters(contractsMap: ChainMap<Chain, Contracts>) {
```
Comment (asaj) [inline typescript/deploy/src/router/deploy.ts:42]: enrollRemoteRouters?

**Review by asaj — APPROVED** [review]

File: typescript/deploy/src/router/deploy.ts [hunk 3]
```diff
@@ -57,6 +55,24 @@ export abstract class AbacusRouterDeployer<
         }
       }),
     );
+  }
+
+  async transferOwnership(contractsMap: ChainMap<Chain, Contracts>) {
+    this.logger(`Transfer Ownership of routers to owner ...`);
+    await promiseObjAll(
+      objMap(contractsMap, async (chain, contracts) => {
```
Comment (jmrossy) [inline typescript/deploy/src/router/deploy.ts:63]: The (small) problem with this objMap util we seem to use in many places is that it's not immediately evident if it's iterating on the object values or keys

**Review by jmrossy — APPROVED** [review]

File: typescript/deploy/src/router/deploy.ts [hunk 4]
```diff
@@ -57,6 +55,24 @@ export abstract class AbacusRouterDeployer<
...
+          this.configMap[chain].owner,
+        );
+      }),
+    );
+  }
+
+  async deploy() {
+    const contractsMap = await super.deploy();
+
+    this.enrollRemoteRouters(contractsMap);
```
Comment (yorhodes) [inline typescript/deploy/src/router/deploy.ts:74]: await?

File: typescript/deploy/src/router/deploy.ts [hunk 5]
```diff
@@ -57,6 +55,24 @@ export abstract class AbacusRouterDeployer<
...
+    this.transferOwnership(contractsMap);
```
Comment (yorhodes) [inline typescript/deploy/src/router/deploy.ts:75]: await?

**Pull request merged**